en/dev_log/log_encryption.md

```diff
@@ -0,0 +1,179 @@
+sudo apt-get install libncurses-dev flex bison openssl libssl-dev dkms libelf-dev libudev-dev libpci-dev libiberty-dev autoconf
+```
+
+Now, in PX4, run the normal `make` command you would use to build the board you are targeting, but add "menuconfig" at the end of it.
+As an example:
+
+```sh
+make px4_fmu-v5_cryptotest menuconfig
+```
+
+Now go to "Crypto API".
+
+![KConfig Crypto API Main Menu Option](../../assets/hardware/kconfig-crypto-1.png)
+
+Check "Crypto API Support", "Blake2s hash algorithm", "Entropy pool and strong random number generator" and "Use interrupts to feed timing randomness to entropy pool".
+
+![KConfig Crypto Options Set](../../assets/hardware/kconfig-crypto-2.png)
+
+::: tip
+Some of these options can be tweaked if desired.
+:::
+
+After enabling encryption settings in kconfig, you may now build and test.
+
+## Download Encrypted Log Files From QGroundControl
+
+::: info
+When you download log files off of QGroundControl, both the encrypted log and the symmetric key will have the ".ulg" suffix.
+You can tell which one is the symmetric key by the file size.
+The symmetric key is normally pretty small (around 300 bytes).
+:::
+
+The process for downloading log files is the same, except you will download two separate files per log.
+Each time you fly, a log file, and a symmetric key are generated.
+The only way to tell them apart is by file size.
+
+As normal, go to **Analyze Tools -> Log Download** and select the log you wish to download.
+
+Make sure to download both the encrypted log and symmetric key that were generated.
+
+![QGroundControl ULog Download](../../assets/qgc/analyze/encrypted_log.png)
+
+This photo shows both a generated symmetric key and the associated encrypted ULog file.
+
+### Decrypt ULogs
+
+Before you can analyze your encrypted logs, you will need to decrypt them.
+There is a Python script that can be used to decrypt logs in Tools/decrypt_ulog.py.
+
+decrypt_ulog.py takes 3 arguments:
+
+1. The encrypted `.ulogc` file.
+2. The symmetric key `.ulogk` file.
+3. The decryption key (the RSA2048 `.pem` private key which is used to unwrap the `.ulogk` file).
+
+```sh
+usage: decrypt_ulog.py [-h] [ulog_file] [ulog_key] [rsa_key]
+
+CLI tool to decrypt an ulog file
+
+positional arguments:
+  ulog_file   .ulog file
+  ulog_key    .ulogk, encrypted key
+  rsa_key     .pem format key for decrypting the ulog key
+
+optional arguments:
+  -h, --help  show this help message and exit
+
+```
+
+As an example:
+
+```sh
+python3 decrypt_ulog.py \
+/home/john/Downloads/log_24_2024-10-6-23-39-50.ulg \
+/home/john/Downloads/log_23_2024-10-6-23-39-48.ulg \
+new_keys/private_key.pem
+```
+
+There will be no printed output on a successful decryption, and a new file is created with the ".ul" suffix instead of ".ulg".
+Rename this back to a .ulg file and it is now ready for flight review!
+
+### Generate Your Own Keys
+
+In a production environment, you should your own generated keys rather than using the example keys.
```

Comment:
Ah, so there are example keys. Where can they be found?

Reply:
Just pushed up change that clarifies this